docs(deepseek): clarify model names (V3, R1) in docstring

kelvinvelasquez-SDE opened the pull request:
## Description

Improves documentation for the DeepSeek integration by clarifying which model names map to which DeepSeek releases.

## Changes

- **Modified `chat_models.py`**: Updated docstring to clarify that `'deepseek-chat'` uses DeepSeek-V3 and `'deepseek-reasoner'` uses DeepSeek-R1.

## Context

This is a documentation-only change. Initially, this PR attempted to add a `deepseek-v3` profile, but after reviewer feedback, I learned that:
1. `deepseek-v3` is not a valid API model ID (the correct ID is `deepseek-chat`)
2. Profile data is auto-generated from models.dev and shouldn't be manually edited

The PR was updated to focus solely on improving the docstring for user clarity.

kelvinvelasquez-SDE committed:
feat(providers): add explicit DeepSeek-V3 profile and docs

mdrxy commented on libs/partners/deepseek/langchain_deepseek/chat_models.py:
```diff
@@ -56,7 +56,8 @@ class ChatDeepSeek(BaseChatOpenAI):
     Key init args — completion params:
         model:
-            Name of DeepSeek model to use, e.g. `'deepseek-chat'`.
+            Name of DeepSeek model to use, e.g. "deepseek-chat", "deepseek-v3",
```
```suggestion
            Name of DeepSeek model to use, e.g. `'deepseek-chat'`, `'deepseek-v3'`,
```

mdrxy commented on libs/partners/deepseek/langchain_deepseek/chat_models.py:
```diff
@@ -56,7 +56,8 @@ class ChatDeepSeek(BaseChatOpenAI):
         model:
-            Name of DeepSeek model to use, e.g. `'deepseek-chat'`.
+            Name of DeepSeek model to use, e.g. "deepseek-chat", "deepseek-v3",
+            or "deepseek-reasoner".
```
```suggestion
            or `'deepseek-reasoner`'.
```

mdrxy committed:
Apply suggestion from @mdrxy

mdrxy committed:
Update libs/partners/deepseek/langchain_deepseek/chat_models.py

mdrxy committed:
Merge branch 'master' into feat/deepseek-v3-support

mdrxy committed:
Merge branch 'master' into feat/deepseek-v3-support

kelvinvelasquez-SDE committed:
refactor(deepseek): improve docs, remove invalid profile

- Update docstring to clarify that 'deepseek-chat' uses DeepSeek-V3
  and 'deepseek-reasoner' uses DeepSeek-R1
- Remove 'deepseek-v3' profile from _profiles.py (not a valid API model ID)
- Remove corresponding test for non-existent profile

This follows reviewer feedback: model profiles are auto-generated from
models.dev, so manual entries would be overwritten.

kelvinvelasquez-SDE commented:
I've updated the PR based on the feedback:

**Changes made:**
- âœ… Removed the `deepseek-v3` profile from `_profiles.py` (since it's not a valid API model ID and would be overwritten by auto-generation)
- âœ… Removed the corresponding unit test
- âœ… Improved the docstring to clarify that `deepseek-chat` uses DeepSeek-V3 under the hood, and `deepseek-reasoner` uses DeepSeek-R1

The PR now focuses solely on documentation improvement, which should be safe and helpful for users who may not know which model name maps to which DeepSeek release.

Ready for re-review when you have a moment. Thanks again for the guidance! ðŸ™

mdrxy commented:
I think it may be best to leave this ambiguous so as not to expand the surface area of what is dependent on upstream sources to maintain
